src/MinterGateway.sol

```diff
@@ -866,13 +873,22 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
      * @param newTimestamp_ The timestamp of the collateral update.
      */
     function _updateCollateral(address minter_, uint240 amount_, uint40 newTimestamp_) internal {
-        uint40 lastUpdateTimestamp_ = _minterStates[minter_].updateTimestamp;
+        MinterState storage minterState_ = _minterStates[minter_];
 
-        // MinterGateway already has more recent collateral update
+        uint40 lastUpdateTimestamp_ = minterState_.updateTimestamp;
+
+        // Revert if minter's last collateral update is more recent.
         if (newTimestamp_ <= lastUpdateTimestamp_) revert StaleCollateralUpdate(newTimestamp_, lastUpdateTimestamp_);
 
-        _minterStates[minter_].collateral = amount_;
-        _minterStates[minter_].updateTimestamp = newTimestamp_;
+        uint40 latestProposedRetrievalTimestamp_ = minterState_.latestProposedRetrievalTimestamp;
+
+        // Revert if minter's last created retrieval proposal is more recent.
+        if (newTimestamp_ <= latestProposedRetrievalTimestamp_) {
+            revert ObsoleteCollateralUpdate(newTimestamp_, latestProposedRetrievalTimestamp_);
```

Comment:
What is the difference between `StaleCollateralUpdate` and `ObsoleteCollateralUpdate`?
Both errors log the same kind of value (two timestamps) and it is not really clear which one is about `lastUpdateTimestamp_` or `latestProposedRetrievalTimestamp_`.
I think keeping only `StaleCollateralUpdate` would make more sense.
```suggestion
            revert StaleCollateralUpdate(newTimestamp_, latestProposedRetrievalTimestamp_);
```